src/PollingBlockTracker.ts

```diff
@@ -118,37 +117,23 @@ export class PollingBlockTracker
     // return if available
     if (this._currentBlock) {
       return this._currentBlock;
+    } else if (this.#pendingLatestBlock) {
```

Comment:
Ah, good catch here. I was thinking of also using a deferred promise and keeping it in a private field, but it seemed that calling `getLatestBlock` multiple times without waiting would overwrite that field. This is a good solution.